domain/src/main/java/life/qbic/usermanagement/repository/UserRepository.java

```diff
@@ -0,0 +1,42 @@
+package life.qbic.usermanagement.repository;
+
+import java.util.Optional;
+import life.qbic.usermanagement.User;
+
+public class UserRepository {
+
+  private static UserRepository INSTANCE;
+
+  private final UserDataStorage dataStorage;
+
+  public static UserRepository getInstance(UserDataStorage dataStorage) {
+    if (INSTANCE == null) {
+      INSTANCE = new UserRepository(dataStorage);
+    }
+    return INSTANCE;
+  }
+  protected UserRepository(UserDataStorage dataStorage) {
+    this.dataStorage = dataStorage;
+  }
+
+  public Optional<User> findByEmail(String email) {
+    var matchingUsers = dataStorage.findUsersByEmail(email);
+    if (matchingUsers.size() > 1) {
+      throw new RuntimeException("More than one user entry with the same email exists!");
+    }
+    return Optional.empty();
+  }
```

Comment:
You actually never return a user here, right?

Reply:
ha, nice spot!